src/vectors.S

```diff
@@ -0,0 +1,92 @@
+.section .text.init
+.global _run_guest_vector,_guest_exit_vector
+
+// t6 -> cached returned value
+// t5 -> cached sstatus
+// restore vectors for guest
+// i.e. memory to main state
+.align 2
+_run_guest_vector:
+    csrr t5, sstatus
+
+    // if vs != 0, restore
+    // enable vectors
+
+    ld t1, ({guest_vl})(a0)
+    ld t2, ({guest_vtype})(a0)
+    vsetvl t0,t1,t2
+
+    // check that vtype is ok?
+
+    ld t0, ({guest_vstart})(a0)
+    csrw vstart, t0
+
+    ld t0, ({guest_vcsr})(a0)
+    csrw vcsr, t0
+    
+
+
+    addi t3, a0, {guest_v0}
+    vl8r.v  v0, (t3)
+    addi t3, a0, {guest_v8}
+    vl8r.v  v8, (t3)
+    addi t3, a0, {guest_v16}
+    vl8r.v  v16, (t3)
+    addi t3, a0, {guest_v24}
+    vl8r.v  v24, (t3)
+
+    csrw sstatus, t5
+    ret
+
+// t6 -> cached returned value
+// t1 -> cached sstatus
+// save vectors for guest
+// i.e. from guest to mem
+.align 2
+_guest_exit_vector:
+    csrr  t1, sstatus
+    li    t0, {sstatus_vs_dirty}
+    and   t2, t1, t0
+    bne   t2, t0, _guest_exit_vector_return
+    not   t0, t0
+    and   t1, t1, t0
+    li    t0, {sstatus_vs_clean}
+    or    t1, t1, t0
+
+    csrrs  t4, vcsr, zero
```

Comment:
nit: Using the `csrr` pseudo-instruction is more readable than using `csrrs`.

Reply:
fixed